src/core.js

```diff
@@ -478,7 +478,13 @@ jQuery.each("Boolean Number String Function Array Date RegExp Object Error".spli
 });
 
 function isArraylike( obj ) {
-	var length = obj.length,
+
+	// Support: iOS 8.2 (not reproducible in simulator)
+	// `in` check used to prevent JIT error (gh-2145)
+	// hasOwn isn't used here due to false negatives
+	// regarding Nodelist length in IE
+	// obj is casted to Object to avoid type errors (gh-2185)
+	var length = "length" in Object( obj ) && obj.length,
```

Comment:
I just double-checked, and removing this `Object` won't break anything. [`jQuery.makeArray`](https://github.com/timmywil/jquery/blob/2.1-stable/src/core.js#L353) already casts its input, and [`jQuery.each`](https://github.com/timmywil/jquery/blob/2.1-stable/src/core.js#L296) and [`jQuery.map`](https://github.com/timmywil/jquery/blob/2.1-stable/src/core.js#L408) already fail on strings: http://jsfiddle.net/j2qyduL9/ (which covers all uses of this function).